src/clj_kondo/impl/analysis.clj

```diff
@@ -96,3 +96,10 @@
                        :filename filename
                        :lang (when (= :cljc (:base-lang ctx)) (:lang ctx))
                        :id (:id binding)))))
+
+(defn reg-keyword-usage! [{:keys [:analysis] :as ctx} filename usage]
```

Comment:
Minor nitpick, but when using only one key it's generally better for performance not to use destructuring. Just use `(when-let [analysis (:analysis ctx)] ...)`